### PR TITLE
items: fix walk to pickups

### LINKED ISSUE #834: Overzealous walk to pickup
Lara shuffles her feet even when she's standing literally on the item she's about to pick up. Let's check if we broke something recently related to this. If not, let's expand the walk activation radius to something more sensible, and check the radii for this feature to work in general.

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Resolves #834

### CODE REVIEW

File: src/game/items.c [hunk 1]
```diff
@@ -416,28 +388,95 @@ void Item_AlignPosition(
...
+        if (velocity >= dist) {
+            item->pos.x = dst_pos.x;
+            item->pos.y = dst_pos.y;
+            item->pos.z = dst_pos.z;
+        } else {
+            item->pos.x += (dx * velocity) / dist;
+            item->pos.y += (dy * velocity) / dist;
+            item->pos.z += (dz * velocity) / dist;
+        }
+    }
```
Comment (walkawayy) [inline src/game/items.c:430]: Why the floating `{` `}` around this chunk of code? Because `dx` and `dz` are reused later?

**Review by walkawayy — COMMENTED** [review]
I like the changes. This needs to be tested with switches, pickups, Scion pieces, and more (?) with the option turned on and off.

File: src/game/items.c (hunk 1 above)
Comment (rr-) [reply src/game/items.c:430]: Yes. Is it awful?

Comment (walkawayy) [reply src/game/items.c:430]: It could possibly be confusing, but idk better names off the top of my head lol.

**Comment by walkawayy** [comment]
| Target | Walk to Pickups | Result |
| ------- | ------- | ------- |
| Switch | Yes | :heavy_check_mark: |
| Switch | No | :heavy_check_mark: |
| Item | Yes | :heavy_check_mark: |
| Item | No | :heavy_check_mark: |
| UW Item | Yes | :heavy_check_mark: |
| UW Item | No | :heavy_check_mark: |
| Scion Altar | Yes | :heavy_check_mark: |
| Scion Altar | No | :heavy_check_mark: |
| Pierre Scion | Yes | :heavy_check_mark: |
| Pierre Scion | No | :heavy_check_mark: |

**Comment by walkawayy** [comment]
LGTM and tests look good. Last question I had was about `Pickup_CollisionControlled`. Did you decide on keeping the item rot changes code?

```
int16_t rotx = item->pos.x_rot;
int16_t roty = item->pos.y_rot;
int16_t rotz = item->pos.z_rot;
...
item->pos.x_rot = rotx;
item->pos.y_rot = roty;
item->pos.z_rot = rotz;
```

**Review by walkawayy — APPROVED** [review]

**Comment by rr-** [comment]
> Did you decide on keeping the item rot changes code?

Sorry but I'm not sure what this code's about, can I have more context?

**Review by lahm86 — APPROVED** [review]
LGTM. Tested some obscure pickup locations as well, such as on slopes or where Lara has to clip through cracks (hard/glitched secrets for TR-Rando in mind), and it's all working as expected.

**Comment by walkawayy** [comment]
> > Did you decide on keeping the item rot changes code?
> 
> Sorry but I'm not sure what this code's about, can I have more context?

in `pickup.c`, both `Pickup_Collision` and `Pickup_CollisionControlled` modify `item->pos.n_rot`s. I don't think it messes with anything, especially because pickups are sprites in OG TR1 that are always rotating, but I just wanted to bring it up since you mentioned wanting to be conscious of the state. Only `Pickup_CollisionControlled` restores the values since it was ported from newer TR5.

**Comment by rr-** [comment]
I see. Yes, let's keep it as-is, as it's currently working, and I don't have the energy necessary to make it nicer.

**Pull request merged**